### ISSUE TITLE
ROKU ISSUE

### ISSUE THREAD

**Issue by Jag4343** [dtankdempse/free-iptv-channels#27]
Using Tivimate and Roku channels are updating and EPG is now working but all of the channels are not working and getting a data source exception error. Can this be fixed? Thanks

**Comment by pucknbass** [comment]
yeah roku give a playlist but nor stream url..... i think they now switched stuff everyday

**Comment by Jag4343** [comment]
Does anybody know the NEW working URL for Roku?Sent from my iPhoneOn Dec 13, 2024, at 10:50 AM, pucknbass ***@***.***> wrote:﻿
yeah roku give a playlist but nor stream url..... i think they now switched stuff everyday

**Comment by clseibold** [comment]
mjh no longer provides m3u urls for anything, and it looks like this repository's code is still trying to use mjh's Roku channels list as if it still provided these m3u urls. @dtankdempse The `handleRoku` function needs to be updated to account for this. However, I don't know if there's any other service that can be used for linking the playlists.

**Comment by dtankdempse** [comment]
Once Matt addresses the issue with the Roku, it should start working again. Although his repository no longer lists M3U files, it still provides JSON files, which this repository converts into M3U playlists.

**Comment by Jag4343** [comment]
Hi Reuben. I’m still not seeing a direct link for thetvapp.  Are you able to make one available? Sent from my iPhoneOn Jan 2, 2025, at 10:48 AM, Reuben ***@***.***> wrote:﻿
Once Matt addresses the issue with the Roku, it should start working again. Although his repository no longer lists M3U files, it still provides JSON files, which this repository converts into M3U playlists.

**Comment by dtankdempse** [comment]
@Jag4343 A message was sent to you on discord.

**Comment by matthuisman** [comment]
i believe this can be closed as a duplicate of https://github.com/dtankdempse/free-iptv-channels/issues/26
